platforms/core-runtime/launcher/src/main/java/org/gradle/tooling/internal/provider/ProviderStartParameterConverter.java

```diff
@@ -55,7 +56,7 @@ private List<TaskExecutionRequest> unpack(final List<InternalLaunchable> launcha
         return requests;
     }
 
-    public StartParameterInternal toStartParameter(ProviderOperationParameters parameters, BuildLayoutResult buildLayout, AllProperties properties) {
+    public StartParameterInternal toStartParameter(ProviderOperationParameters parameters, BuildLayoutResult buildLayout, AllProperties properties, Map<String, String> environmentVariables) {
```

Comment:
Not now, but should AllProperties be carrying environment variables too?

Reply:
I created #35850. There are things to clarify since there is `AllProperties.merge` so I don't think it needs to happen right now.